src/symbolize/gimli/xcoff.rs

```diff
@@ -0,0 +1,184 @@
+use super::mystd::ffi::{OsStr, OsString};
+use super::mystd::os::unix::ffi::OsStrExt;
+use super::mystd::str;
+use super::{gimli, Context, Endian, EndianSlice, Mapping, Path, Stash, Vec};
+use alloc::sync::Arc;
+use core::ops::Deref;
+use object::read::archive::ArchiveFile;
+use object::read::xcoff::{FileHeader, SectionHeader, XcoffFile, XcoffSymbol};
+use object::Object as _;
+use object::ObjectSection as _;
+use object::ObjectSymbol as _;
+use object::SymbolFlags;
+
+#[cfg(target_pointer_width = "32")]
+type Xcoff = object::xcoff::FileHeader32;
+#[cfg(target_pointer_width = "64")]
+type Xcoff = object::xcoff::FileHeader64;
+
+impl Mapping {
+    pub fn new(path: &Path, member_name: &OsString) -> Option<Mapping> {
+        let map = super::mmap(path)?;
+        Mapping::mk(map, |data, stash| {
+            if member_name.is_empty() {
+                Context::new(stash, Object::parse(data)?, None, None)
+            } else {
+                let archive = ArchiveFile::parse(data).ok()?;
+                for member in archive.members() {
+                    if let Ok(member) = member {
+                        let name = OsStr::from_bytes(member.name());
+                        if name == member_name {
```

Comment:
While it's not _pretty_ either way, this should look something more like
```suggestion
                for member in archive.members()
                                     .filter_map(|m| m.ok())
                                     .filter(|m| OsStr::from_bytes(m.name()) == member_name) {
```
to eliminate the severe rightward drift. Probably rustfmt disagrees with exactly how I wrote it, even after rebalancing the braces.